### PR TITLE
yixiao_protectedroute_unittest

### PR DESCRIPTION
**Pull request by JYXiao-2021**
# Description
Create the unit test for the ProtectedRoute component to confirm whether it working as intended.

## Main changes explained:
- Create file ProtectedRoute.test.js to do the unit test.

## How to test:
1.check into current branch
2.do npm install to run this PR locally
3.run npm test ProtectedRoute.test.js

## Screenshots or videos of changes:
image

### CODE REVIEW

**Review by AaronPersaud — DISMISSED** [review]
Hi @JYXiao-2021, I've tested and all the tests have passed! Good job.

image

**Comment by iTvX** [comment]
It passed the test. Good job!

CleanShot 2024-02-25 at 01 52 16@2x

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I tested your PR and it passed all the test cases. Please refer the screenshot attached.

PR 1993

**Review by ChengyanOo — DISMISSED** [review]
All test passed, great work!
Screenshot 2024-02-26 at 17 20 06

**Review by iTvX — CHANGES_REQUESTED** [review]
The unit test should be placed in the folder named "__tests__." Please create a "__test__" folder and place the unit test under this folder.

CleanShot 2024-02-26 at 20 15 56@2x

**Review by ShengjieMao — DISMISSED** [review]
All tests passed and the components and props are comprehensively considered. As others have indicated, it might be better to organize the

 files under a test dir.

**Review by XiaohanMeng — DISMISSED** [review]
Hi, all 6 test cases passed! Great job! 
1993_result

**Review by skaush21 — APPROVED** [review]
Ran npm test, all tests passed.
Screen Shot 2024-02-28 at 8 19 08 PM

File: src/components/common/ProtectedRoute/ProtectedRoute.test.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    ...render(ui, { wrapper: Wrapper }),
+    history,
+  };
+};
+
+const mockStore = configureStore([thunk]);
+
+// Mock components for testing
+const LoginComponent = () => <div>Login Page</div>;
+const DashboardComponent = () => <div>Dashboard Page</div>;
```
Comment (FyisFe) [inline src/components/common/ProtectedRoute/ProtectedRoute.test.js:35]: Those two components are declared but not used.

**Review by FyisFe — DISMISSED** [review]
All tests passed. 

image

**Comment by dhairya-mehta** [comment]
Hey verified the Pull Request, good job!
image

**Review by scoopstroop92 — CHANGES_REQUESTED** [review]
Good job on the test cases. Kindly update the path 
image

**Review by Pandani07 — DISMISSED** [review]
All the test cases passed! Great work

**Comment by JYXiao-2021** [comment]
> Good job on the test cases. Kindly update the path image

update the path, please review

**Review by iTvX — APPROVED** [review]

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR and all the test cases passed!

Week 28 - PR3

**Review by scoopstroop92 — APPROVED** [review]
Looks good !

**Pull request merged**